_posts/2020-11-19-testy-e2e-cypress.md

```diff
@@ -0,0 +1,131 @@
+
+Pierwszy scenariusz będzie polegał na wejście na stronę, poczekaniu aż się załaduje i zweryfikowaniu kilku elementów, która potwierdzą nam, że portal jest w pełni działający.
+
+Drugi scenariusz, nieco bardziej rozbudowany, dokona kilku interakcji z blogiem.
+
+# Pierwszy test
+
+W katalogu `integration` stwórzmy nowy plik o nazwie `blog.spec.js`, przejdźmy do jego edycji i uzupełnijmy go o następującą treść:
+```javascript
+describe('Blog Consdata Tech', () => {
+    it('should check page fully loaded', () => {
+        // ..
+    });
+})
+```
+Tak jak wcześniej wspomniałem, jeśli ktoś miał wcześniej styczność z testami w JS, wszystko będzie dla niego jasne.
+
+Idąc od początku, `describe` nazywa nam całościowy kontekst, którego będą dotyczyć poszczególne scenariusze. `it` rozpoczyna konkretny przypadek testowy i to wewnątrz niego będziemy implementować właściwą logikę.
+
+W tym momencie możemy uruchomić dashboard Cypressa. Zobaczymy w nim, że pojawił się nasz plik. Po kliknięciu w niego uruchomi się przeglądarka oraz scenariusz testowy:
+<div class="img-with-legend">
+<img alt="Cypress scenariusz testowy" src="/assets/img/posts/2020-11-19-testy-e2e-cypress/scenariusz-testowy-cypress.png" />
+<span class="img-legend">Rys. 3. Uruchomiony scenariusz w Cypress</span>
+</div>
+Zostawmy przeglądarkę i dashboard włączony w tle - Cypress nasłuchuje na zmiany i od razu je wykonuje. Dzięki temu mamy ciągłą pętlę zwrotną z informacją czy to co tworzymy przynosi oczekiwane efekty.
+
+Wracając do tworzenia testu, jak wynika z nazwy `should check page fully loaded`, pierwszy z nich będzie sprawdzał czy strona na którą wchodzimy poprawnie się załadowała. Dodajmy ciało do pustej metody:
+```javascript
+cy.visit('https://blog.consdata.tech/').then(() =>{
+    cy.get('.header-logo').should('be.visible');
+    cy.get('footer').should('be.visible');
+});
+```
+Jak widać, napisany kod jest właściwie samoopisujący się. W pierwszej linii odwiedzamy podany adres, następnie, gdy to się zadzieje, szukamy w DOM elementu o klasie `header-logo` i sprawdzamy czy jest widoczny. Operacje powtarzamy dla elementu `footer`. W przeglądarce uświadczy nas poniższy obrazek:
+<div class="img-with-legend">
+<img alt="Dashboard Cypress" src="/assets/img/posts/2020-11-19-testy-e2e-cypress/sukces%20scenariusz%20pierwszy%20cypress.png" />
+<span class="img-legend">Rys. 4. Scenariusz testowy zakończony powodzeniem</span>
+</div>
+
+# Interakcje
+
+Twórcy Cypressa przygotowali nam zestaw metod, które ułatwią nam wykonywanie operacji na stronie. W skład tych poleceń wchodzi m.in. `click()`, `type()`, `select()` czy `check()`.
+
+Jeszcze zanim przejdziemy do implementacji drugiego scenariusza, trzeba zauważyć, że każdy z nich będzie zaczynał się od tej samej akcji - otworzenia strony. Możemy wynieść ten kawałek kodu to do metody `beforeEach()`, która jest uruchamiana przed każdym testem:
```

Comment:
jakieś "to" się tutaj zapodziało :)
==> Możemy wynieść ten kawałek kodu do metody